packages/dynamic_layouts/README.md

```diff
@@ -13,25 +13,138 @@ and the Flutter guide for
 [developing packages and plugins](https://flutter.dev/developing-packages).
 -->
 
-TODO: Put a short description of the package here that helps potential users
-know whether this package might be useful for them.
+<!-- TODO(DavBot02): Put a short description of the package here that helps potential users
+know whether this package might be useful for them.-->
 
 ## Features
+This package provides support for multi sized tiles and different layouts.
+Currently the layouts that are implemented in this package are `Stagger` and
+`Wrap`.
 
-TODO: List what your package can do. Maybe include images, gifs, or videos.
+The following are some demos of how each of the grids look.
+
+A stagger grid demo:
+
+<!-- TODO(snat-s): Add stagger video demo -->
+
+A wrap demo:
+
+<!-- TODO(snat-s): Add wrap video demo -->
+
+### Stagger Features
+
+`DynamicGridView` is a subclass of `GridView` and gives access
+to the `SliverGridDelegate`s that are already implemented in the Flutter
+Framework. Some `SliverGridDelegate`s are `SliverGridDelegateWithMaxCrossAxisExtent` and
+`SliverGridDelegateWithFixedCrossAxisCount`. This layout can be used with
+`DynamicGridView.stagger`.
+
+### Wrap Features
+
+The Wrap layout is able to do runs of different widgets and adapt accordingly with
+the sizes of the children. It can leave spacing with `mainAxisSpacing` and
+`crossAxisSpacing`.
+
+Having different sizes in only one of the axis is possible by
+changing the values of `childCrossAxisExtent` and `childMainAxisExtent`. These
+values by default are ignored, but changing `childCrossAxisExtent` to be
+100 pixels, will make all of the children 100 pixels in the main axis.
+This layout can be used with `DynamicGridView.wrap` and with
+`DynamicGridView.builder` and `SliverGridDelegateWithWrapping` as the delegate.
 
 ## Getting started
 
-TODO: List prerequisites and provide or point to information on how to
-start using the package.
+<!-- TODO(DavBot02): List prerequisites and provide or point to information on how to start using the package. -->
 
 ## Usage
 
-TODO: Include short and useful examples for package users. Add longer examples
-to `/example` folder.
+Use `DynamicGridView`s to access this layouts.
+`DynamicGridView` has some constructors that have specific `SliverGridDelegate`s like
+`DynamicGridView.wrap` and `DynamicGridView.stagger`. For a more efficient option
+`DynamicGridView.builder` works the same as `GridView.builder`.
```

Comment:
This could be a bit more specific.

.wrap and .stagger use SliverChildListDelegate, and .builder uses SliverChildBuilderDelegate